src/main/java/com/palehorsestudios/alone/PlayerInput.java

```diff
@@ -0,0 +1,34 @@
+package com.palehorsestudios.alone;
+
+import java.util.Scanner;
+
+public final class PlayerInput { // singleton class to manage all player inputs
+  private static PlayerInput refToPlayerInput = null;
+  private Scanner scanner;
+  public PlayerInput() {
```

Comment:
I think for this to be a singleton, the constructor needs to be private. That way PlayerInput cannot be instantiated directly. The only way a PlayerInput can get instantiated would be through the public static getInstance method. During our Java training, we also used enums to implement singletons. You might review that topic in our textbook.